packages/core/src/components/icon/icon.tsx

```diff
@@ -171,7 +171,7 @@ export const Icon: IconComponent = React.forwardRef(function <T extends Element>
                 : size === IconSize.LARGE
                   ? Classes.ICON_LARGE
                   : undefined;
-        return React.createElement(tagName!, {
```

Comment:
I think it makes sense to ensure this value is always a valid element. The alternative would be to do something like a null check and return early, e.g.
```tsx
if (tagName == null) {
    return null;
}
return React.createElement(tagName, { ... });
```

The former will show a blip of the fallback while the icon paths load:

![Screenshot 2024-09-17 at 16 21 42](https://github.com/user-attachments/assets/732760b1-90ff-43c2-8522-4031ab3c754b)

whereas the later will appear and shift in without any placeholder at all:

![Screenshot 2024-09-17 at 16 22 34](https://github.com/user-attachments/assets/fa914129-f15d-4f1a-b344-b9ffe3a5d81b)

Reply:
Any particular reason to use `||` here over `??`?

Reply:
Hm, I suppose the only main difference is that `||` will use `"span"` as the default for _any_ falsy value, including `""`, `0`, and `false`. If the user is consuming this component in TypeScript, there is functionally no difference since `tagName` is either type of`React.JSX.IntrinsicElements` or `null`.

Do we care about this edge case for non-TypeScript consumers?

Reply:
Yeah okay, that makes sense. I think its fair to keep this for non-TS consumers.